src/cdomains/apron/linearTwoVarEqualityDomain.apron.ml

```diff
@@ -16,96 +16,113 @@ open VectorMatrix
 
 module Mpqf = SharedFunctions.Mpqf
 
-module Equality = struct
+module Rhs = struct
   (* (Some i, k) represents a sum of a variable with index i and the number k.
      (None, k) represents the number k. *)
   type t = (int option * GobZ.t) [@@deriving eq, ord, hash, show]
   let zero = (None, Z.zero)
   let var_zero i = (Some i, Z.zero)
-  let to_int x = Z.to_int @@ snd x
+  let to_int (_,x) = Z.to_int x
+  let show (v, o) = Option.map_default (fun i -> "var_" ^ string_of_int i^ " + ") "" v  ^ Z.to_string o
 end
 
-module EqualitiesArray = struct
-  include Array
-  type t = Equality.t Array.t [@@deriving eq, ord]
+module EqualitiesConjunction = struct
+  module IntMap = BatMap.Make(Int)
 
-  let show m =
-    Array.fold_right (fun k result -> Equality.show k ^ "\n" ^ result) m ""
+  type t = int * ( Rhs.t IntMap.t ref)
```

Comment:
I'm a bit confused what purpose it serves to have a `ref` here?

Reply:
I start to be more and more confused about that myself. I guess the main reason is historic, due to  the former array having been mutable, and me not daring enough to change that character. Also, I think I need re-education wrt. when to use _with , and what that really demands semantically. Lets's do that in person.

Reply:
I have removed ref now